R/sim.R

```diff
@@ -292,18 +292,18 @@ sim <- function (ode = NULL,
     size <- attr(ode, "size")
   }
   if (!is.null(omega)) {
-    if(class(omega) == "matrix") {
+    if(inherits(omega, "matrix")) {
       omega_mat <- omega
     }
-    if(class(omega) %in% c("numeric", "integer")) {
+    if(inherits(omega, "numeric")) {
```

Comment:
it's very unlikely that omega is ever going to be an integer, but if you want to catch them both you can use `mode`:

```suggestion
    if(mode(omega) == "numeric") {
```

mode will always return a logical of length 1.